File: productengineering/persistence/src/main/java/br/com/todeschini/persistence/entities/publico/Filho.java
package br.com.todeschini.persistence.entities.publico;

import br.com.todeschini.domain.metadata.Entidade;
import br.com.todeschini.persistence.entities.enums.TipoFilho;
import br.com.todeschini.persistence.entities.enums.converters.TipoFilhoConverter;
import lombok.*;

import javax.persistence.*;
import java.time.LocalDate;
import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

@Builder
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@EqualsAndHashCode(of = "cdfilho", callSuper = false)
@Entity
@Table(name = "tb_filho")
@Entidade
public class Filho extends AuditoriaInfo {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer cdfilho;
    private String descricao;

    @ManyToOne
    @JoinColumn(name = "cdpai")
    private Pai pai;

    @ManyToOne
    @JoinColumn(name = "cdcor")
    private Cor cor;

    @ManyToOne
    @JoinColumn(name = "cdmedidas")
    private Medidas medidas;

    @ManyToOne
    @JoinColumn(name = "cdroteiro")
    private Roteiro roteiro;

    private String unidadeMedida = "UN";
    private LocalDate implantacao;
    private Double valor;

    @Convert(converter = TipoFilhoConverter.class)
    private TipoFilho tipo;

    @ManyToMany(cascade = CascadeType.REMOVE, fetch = FetchType.LAZY)
    @JoinTable(name = "tb_item_filho",
            joinColumns = @JoinColumn(name = "item_id"),
            inverseJoinColumns = @JoinColumn(name = "filho_id"))
    private Set<Filho> filhos = new HashSet<>();

    @OneToMany(mappedBy = "filho", cascade = CascadeType.REMOVE, fetch = FetchType.LAZY)
    private List<MaterialUsado> materiaisUsados = new ArrayList<>();

    @OneToMany(mappedBy = "filho", cascade = CascadeType.REMOVE, fetch = FetchType.LAZY)
    private List<AcessorioUsado> acessoriosUsados = new ArrayList<>();

    public Filho(Integer cdfilho) {
        this.cdfilho = cdfilho;
    }

    public Filho(Integer cdfilho, Cor cor, Medidas medidas) {
        this.cdfilho = cdfilho;
        this.cor = cor;
        this.medidas = medidas;
    }
}

File: frontend/src/models/filho.ts
import { DAcessorioUsado } from "./acessorioUsado";
import { DCor } from "./cor";
import { DSituacaoEnum } from "./enums/situacao";
import { DTipoFilhoEnum } from "./enums/tipoFilho";
import { DMaterialUsado } from "./materialUsado";
import { DMedidas } from "./medidas";
import { DPai } from "./pai";
import { DRoteiro } from "./roteiro";

export type DFilho = {
    codigo: number;
    descricao: string;
    pai: DPai;
    cor: DCor;
    medidas: DMedidas;
    roteiro: DRoteiro;
    unidadeMedida: string;
    implantacao: Date;
    valor: number;
    tipo: keyof typeof DTipoFilhoEnum;
    situacao: keyof typeof DSituacaoEnum;
    filhos: DFilho[];
    materiaisUsados: DMaterialUsado[];
    acessoriosUsados: DAcessorioUsado[];
};